core/lib/spree/core/controller_helpers/auth.rb

```diff
@@ -48,6 +48,12 @@ def current_ability
           @current_ability ||= Spree::Ability.new(try_spree_current_user)
         end
 
+        # @deprecated Use #redirect_back provided in Rails 5+
+        def redirect_back_or_default(default)
+          redirect_to(session["spree_user_return_to"] || default)
+          session["spree_user_return_to"] = nil
+        end
```

Comment:
We use Rails' deprecation system, so with this and the other deprecated method you can do something like:

```ruby
deprecate redirect_back_or_default: "Please use redirect_back provided in Rails 5+ instead", deprecator: Spree::Deprecation
```